modal/runner.py

```diff
@@ -229,7 +254,7 @@ async def _run_app(
 ) -> AsyncGenerator[_App, None]:
     """mdmd:hidden"""
     if environment_name is None:
-        environment_name = config.get("environment")
+        environment_name = typing.cast(str, config.get("environment"))
```

Comment:
Since we own the config object here I wonder if it would be better to make it have type safe null behavior, i.e. make it more similar to protobuf null semantics than `dict.get`. Type safety around the environment configuration in general is a big headache, not sure whether casting makes it better or worse.

Reply:
I think this is the first time I use cast, I like to avoid it... But it helps with type restrictions without breaking in user code if the assumption is (lightly) wrong like an assertion would 😬 I agree, it would be very nice to make our config object type safe!

Reply:
Yeah I have sort of started preferring using null string defaults and keeping things typed as `str` rather than `str | None` and dealing with the risk of assertion errors (or runtime `TypeError`) but it's tricky.